kratos/geometries/brep_point.h

```diff
@@ -0,0 +1,352 @@
+    {
+        BaseType::operator=( rOther );
+        mLocalCoordinates = rOther.mLocalCoordinates;
+        mpBackgroundGeometry = rOther.mpBackgroundGeometry;
+        return *this;
+    }
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    typename BaseType::Pointer Create( PointsArrayType const& ThisPoints ) const override
+    {
+        return typename BaseType::Pointer( new BrepPoint( ThisPoints ) );
+    }
+
+    ///@}
+    ///@name Access to Geometry Parts
+    ///@{
+
+    /**
+    * @brief This function returns the pointer of the background geometry
+    *        of this point. Accesseable via BACKGROUND_GEOMETRY_INDEX.
+    * @param Index: only possible with BACKGROUND_GEOMETRY_INDEX.
+    * @return pointer of geometry, corresponding to the index.
+    */
+    GeometryPointer pGetGeometryPart(IndexType Index) override
+    {
+        const auto& const_this = *this;
+        return std::const_pointer_cast<GeometryType>(
+            const_this.pGetGeometryPart(Index));
+    }
+
+    /**
+    * @brief This function returns the pointer of the background geometry
+    *        of this point. Accesseable via BACKGROUND_GEOMETRY_INDEX.
+    * @param Index: only possible with BACKGROUND_GEOMETRY_INDEX.
+    * @return pointer of geometry, corresponding to the index.
+    */
+    const GeometryPointer pGetGeometryPart(IndexType Index) const override
+    {
+        if (Index == BACKGROUND_GEOMETRY_INDEX)
+            return mpBackgroundGeometry;
+
+        KRATOS_ERROR << "Index " << Index << " not existing as geometry part in BrepPoint #"
+            << this->Id() << std::endl;
+    }
+
+    /* @brief only true if BACKGROUND_GEOMETRY_INDEX is used.
+     *        Brep Points have no other geometries.
+     */
+    bool HasGeometryPart(IndexType Index) const override
+    {
+        if (Index == BACKGROUND_GEOMETRY_INDEX)
+            return true;
+
+        return false;
+    }
+
+    ///@}
+    ///@name Geometrical Operations
+    ///@{
+
+    /*
+    * @brief This method maps from dimension space to working space.
+    * @param rResult array_1d<double, 3> with the coordinates in working space
+    * @param LocalCoordinates The local coordinates in dimension space
+    * @return array_1d<double, 3> with the coordinates in working space
+    * @see PointLocalCoordinates
+    */
+    CoordinatesArrayType& GlobalCoordinates(
+        CoordinatesArrayType& rResult,
+        const CoordinatesArrayType& rLocalCoordinates
+    ) const override
+     {
+        mpBackgroundGeometry->GlobalCoordinates(rResult, mLocalCoordinates);
+
+        return rResult;
+    }
+
+    ///@}
+    ///@name Integration Points
+    ///@{
+
+    /* Creates a integration point on the background geometry.
+     * @param return integration points list with one entry.
+     */
+    void CreateIntegrationPoints(
+        IntegrationPointsArrayType& rIntegrationPoints) const override
+    {
+        rIntegrationPoints.resize(1);
```

Comment:
if(rIntegrationPoints.size() < 1)

Reply:
I added it. Thanks for the recommendation.